### PR TITLE
524 - initail changes, added Alumni tab, reusable components

### PR DESCRIPTION
**Pull request by Sukritib13**
Changes -
- Added Alumni tab, used lime colour (The 3rd color in techstart symbol) for LinkedIn icon, Executive tab is still default
- Made code reusable by making it into 3 buttons and using switch statements
- Cleaned up some CSS

### CODE REVIEW

File: src/components/TeamSection/Team.tsx [hunk 1]
```diff
@@ -3,26 +3,38 @@ import Profile from "./Profile";
 import * as S from "./Team.styles";
 import "./Team.styles.ts";
 import { type TeamMember } from "./TeamInformation";
-import { executiveTeam } from "./TeamInformation";
+import { executiveTeam, projectManagers, alumniTeam } from "./TeamInformation";
```
Comment (ParsaKargari) [inline src/components/TeamSection/Team.tsx:6]: I'd combine these two imports - makes it a bit more compact and manageable as they're both from the same file.

File: src/components/TeamSection/teamMembers.json [hunk 2]
```diff
@@ -131,6 +131,20 @@
       "linkedin": "https://www.linkedin.com/in/ammerhamza/"
     }
   ],
+  "alumniTeam": [
```
Comment (ParsaKargari) [inline src/components/TeamSection/teamMembers.json:134]: Probably just wanted to show the UI, but just a reminder to change these to their correct contact :)

**Review by ParsaKargari — CHANGES_REQUESTED** [review]
I like the idea of turning it into a component, but I'd change the UI design back to the slider we had. Also watch out for the responsiveness of the component, it doesn't look like it's responsive for mobile users / small-screen users. 
image
image

File: src/components/TeamSection/teamMembers.json (hunk 2 above)
Comment (Sukritib13) [reply src/components/TeamSection/teamMembers.json:134]: Yes I had made note of that in the description of the PR

File: src/components/TeamSection/Team.tsx [hunk 3]
```diff
@@ -2,27 +2,38 @@ import React from "react";
...
+import { type TeamMember, executiveTeam, projectManagers, alumniTeam } from "./TeamInformation";
 
 type TeamProps = {
   teamMembers: TeamMember[];
   desktopView: boolean;
-  isExec: boolean;
+  activeCategory: "executives" | "projectManagers" | "alumni";
 };
 const Team = (props: TeamProps) => {
+  const baseLength = (() => {
```
Comment (brian-ngyn) [inline src/components/TeamSection/Team.tsx:13]: this should be a memoized value

File: src/components/TeamSection/Profile.tsx [hunk 4]
```diff
@@ -42,9 +42,11 @@ const Profile = (props: ProfileProps) => {
       <S.LinksSection
 
         backgroundColor={
-          props.isExec
+          props.activeCategory === "executives"
             ? SocialMediaColor.ToggleGreen
-            : SocialMediaColor.ToggleBlue
+            : props.activeCategory === "projectManagers"
+            ? SocialMediaColor.ToggleBlue
+            : SocialMediaColor.ToggleYellow
```
[the remark this answers is not in the document]
Comment (Sukritib13) [reply src/components/TeamSection/Profile.tsx]: Addressed!

File: src/components/TeamSection/Team.tsx [hunk 5]
```diff
@@ -2,27 +2,38 @@ import React from "react";
...
+    }
+  })();
   return (
     <S.TeamContainer mobileView={!props.desktopView}>
       {/* eslint-disable-next-line @typescript-eslint/no-explicit-any */}
       {props.teamMembers.map((member: TeamMember) => {
         return (
           <Profile
-            key={props.isExec ? member.id : member.id + executiveTeam.length}
+            key={member.id + baseLength}
```
[the remark this answers is not in the document]
Comment (Sukritib13) [reply src/components/TeamSection/Team.tsx]: Addressed, made it a separate function, memoization on top of it seemed an overkill, but let me know if you still want me to implement it

File: src/components/TeamSection/TeamSection.styles.ts [hunk 6]
```diff
@@ -19,102 +19,56 @@ export const TeamSection = styled.div`
...
+  top: 5px;
+  left: ${({ selectedCategory }) =>
+    selectedCategory === "executives" ? "5px" :
+      selectedCategory === "projectManagers" ? "calc(33.3% + 5px)" : "calc(66.6% + 5px)"};
+  width: calc(33.3% - 10px);
+  height: calc(100% - 10px);
+  background-color: ${({ selectedCategory }) =>
+    selectedCategory === "executives" ? "rgb(77, 214, 168)" :
+      selectedCategory === "projectManagers" ? "rgb(97, 195, 212)" :
+        "#c9e265"};
```
[the remark this answers is not in the document]
Comment (Sukritib13) [reply src/components/TeamSection/TeamSection.styles.ts]: Addressed!

File: src/components/TeamSection/teamMembers.json (hunk 2 above)
Comment (Sukritib13) [reply src/components/TeamSection/teamMembers.json:134]: The headshots for Alumni were added just today so I wasn't able to update it before, but thanks for reminding!

File: src/components/TeamSection/Team.tsx [hunk 7]
```diff
@@ -2,27 +2,40 @@ import React from "react";
...
+import { type TeamMember, executiveTeam, projectManagers, alumniTeam } from "./TeamInformation";
 
 type TeamProps = {
   teamMembers: TeamMember[];
   desktopView: boolean;
-  isExec: boolean;
+  activeCategory: "executives" | "projectManagers" | "alumni";
 };
+
+const getBaseLength = (category: "executives" | "projectManagers" | "alumni") => {
```
Comment (brian-ngyn) [inline src/components/TeamSection/Team.tsx:13]: you should turn this into a type or an enum
 
type myType = "executives" | "projectManagers" | "alumni";

File: src/components/TeamSection/TeamSection.styles.ts [hunk 8]
```diff
@@ -1,5 +1,19 @@
 import styled from "styled-components/macro";
 
+// helper function
```
Comment (ParsaKargari) [inline src/components/TeamSection/TeamSection.styles.ts:3]: I think we've used RGB and hex codes in our codebase plenty of times before, but I think we should move away from directly having these in our code. Would it be possible to store the RGB in a variable and use the variable instead?

**Review by ParsaKargari — DISMISSED** [review]

File: src/components/TeamSection/Team.tsx (hunk 7 above)
Comment (Sukritib13) [reply src/components/TeamSection/Team.tsx]: Addressed!

File: src/components/TeamSection/TeamSection.styles.ts (hunk 8 above)
Comment (Sukritib13) [reply src/components/TeamSection/TeamSection.styles.ts:3]: You are totally right, I had copied them from browser console and forgot to change them into var. Addressed!

File: src/components/TeamSection/Team.tsx [hunk 9]
```diff
@@ -2,27 +2,42 @@ import React from "react";
 import Profile from "./Profile";
 import * as S from "./Team.styles";
 import "./Team.styles.ts";
-import { type TeamMember } from "./TeamInformation";
-import { executiveTeam } from "./TeamInformation";
+import { type TeamMember, executiveTeam, projectManagers, alumniTeam } from "./TeamInformation";
+
+type teamCategory = "executives" | "projectManagers" | "alumni";
```
Comment (brian-ngyn) [inline src/components/TeamSection/Team.tsx:7]: Types should PascalCase

File: src/components/TeamSection/TeamSection.styles.ts [hunk 10]
```diff
@@ -19,102 +33,56 @@ export const TeamSection = styled.div`
...
   margin-top: 2%;
   margin-left: -5%;
+  justify-content: center;
+  align-items: center;
 `;
 
-export const ToggleButtonLabel = styled.label`
-  overflow-y: hidden;
-  display: inline-block;
+export const SliderWrapper = styled.div<{ selectedCategory: "executives" | "projectManagers" | "alumni" }>`
```
Comment (brian-ngyn) [inline src/components/TeamSection/TeamSection.styles.ts:44]: same thing here

File: src/components/TeamSection/TeamSection.styles.ts [hunk 11]
```diff
@@ -19,102 +33,56 @@ export const TeamSection = styled.div`
...
-  @media (max-width: 475px) {
-    height: 0;
-    margin-top: -10px;
+  @media (max-width: 700px) {
+    font-size: 1.2em;
   }
-  font-weight: 500;
 `;
+
+export const SliderPosition = styled.div<{ selectedCategory: "executives" | "projectManagers" | "alumni" }>`
```
Comment (brian-ngyn) [inline src/components/TeamSection/TeamSection.styles.ts:76]: same thing here

File: src/components/TeamSection/TeamSection.tsx [hunk 12]
```diff
@@ -1,39 +1,47 @@
...
-import { executiveTeam, projectManagers } from "./TeamInformation";
+import { executiveTeam, projectManagers, alumniTeam } from "./TeamInformation";
 
 type TeamSectionProps = {
   desktopView: boolean;
 };
 
 const TeamSection = (props: TeamSectionProps) => {
-  const [toggle, setToggled] = useState(false);
+  const [selectedCategory, setSelectedCategory] = useState<"executives" | "projectManagers" | "alumni">("executives");
```
Comment (brian-ngyn) [inline src/components/TeamSection/TeamSection.tsx:12]: same thing here

**Comment by brian-ngyn** [comment]
Looks good, a few more comments + a rebase and should be good to go in

File: src/components/TeamSection/TeamSection.tsx [hunk 13]
```diff
@@ -1,39 +1,48 @@
...
 };
 
 const TeamSection = (props: TeamSectionProps) => {
-  const [toggle, setToggled] = useState(false);
+  const [selectedCategory, setSelectedCategory] = useState<TeamCategory>("executives");
 
-  const getTeamMembers = () => {
-    return toggle ? projectManagers : executiveTeam;
-  };
+  const getTeamMembers = useCallback(() => {
```
Comment (brian-ngyn) [inline src/components/TeamSection/TeamSection.tsx:15]: this should be a memoized value, since it doesn't change unless `selectedCategory` changes

**Review by brian-ngyn — APPROVED** [review]

**Review by Anidion — APPROVED** [review]
Awesome work 🚢

**Pull request merged**